includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -2181,12 +2186,14 @@ public function add_twentytwentyone_mobile_modal() {
 
 		$state_string = str_replace( '-', '_', $amp_lightbox_id );
 
-		AMP_DOM_Utils::add_amp_action( $menu_toggle, 'tap', "{$amp_lightbox_id}.open" );
-		AMP_DOM_Utils::add_amp_action( $menu_toggle, 'tap', "{$body_id}.toggleClass(class=primary-navigation-open,force=true)" );
+		AMP_DOM_Utils::add_amp_action( $menu_open_toggle, 'tap', "{$amp_lightbox_id}.open" );
+		AMP_DOM_Utils::add_amp_action( $menu_open_toggle, 'tap', "{$body_id}.toggleClass(class=primary-navigation-open,force=true)" );
+
+		AMP_DOM_Utils::add_amp_action( $menu_close_toggle, 'tap', "{$amp_lightbox_id}.close" );
+		AMP_DOM_Utils::add_amp_action( $menu_close_toggle, 'tap', "{$body_id}.toggleClass(class=primary-navigation-open,force=false)" );
```

Comment:
I see that this essentially adds two buttons _inside_ a button. I don't think I've seen that done before. It seems to work: https://codepen.io/westonruter/pen/ZEpLeLw

However, for this to work, the `padding` will need to be removed from the `button` to be put instead on the child `span` elements. You can see that the inner `span` (in lime) is a much smaller click target than the button (in red outline):

![image](https://user-images.githubusercontent.com/134745/102003134-4b4e3e00-3cb8-11eb-94ad-4d9eefeabc3e.png)

Also, it will be important to make sure the container button gets `tabindex="-1"` to prevent it from getting focus, since clicking on that outer button will have no resulting action.

```suggestion
		AMP_DOM_Utils::add_amp_action( $menu_close_toggle, 'tap', "{$body_id}.toggleClass(class=primary-navigation-open,force=false)" );
		
		$menu_toggle->setAttribute( 'tabindex', '-1' );
```

Reply:
Ah good catch. Ive applied the suggestion in https://github.com/ampproject/amp-wp/pull/5692/commits/c18b090affcb2935231e5a9da16aa74bd91f4c5e.

Do you think a separate `<button>` to close the menu would be a better solution here? A button within a button does seem a bit odd, now that you bring it up.

Reply:
If this approach works, I think it will probably be safer. The reason is that the button has a `primary-mobile-menu` ID and no class name, so if someone is trying to style it they'd get unexpected results, especially if we deduplicate the ID. In other words, the open button would get the styling but the close button would not.

Reply:
> The reason is that the button has a primary-mobile-menu ID and no class name, so if someone is trying to style it they'd get unexpected results, especially if we deduplicate the ID.

Yea that was my initial concern as well. I haven't noticed any oddities with the current solution so far, and I hope that does fix the bug in Safari.